### ISSUE TITLE
Add Whitelisting

### ISSUE THREAD

**Issue by luckydonald** [qzb/dustman#7]
Whitelisting a tab also automatically specifc urls would be great.

Screenshots from Tab Wrangler.
![bildschirmfoto 2015-10-17 um 17 24 39](https://cloud.githubusercontent.com/assets/2737108/10559571/3eeed390-74f4-11e5-8b50-dc7fa9a2c095.png) 
![bildschirmfoto 2015-10-17 um 17 22 47](https://cloud.githubusercontent.com/assets/2737108/10559574/5fdde172-74f4-11e5-84b0-e7e6cba8654e.png)

**Comment by Sanqui** [comment]
It would be wonderful if you could whitelist tabs simply by double clicking them.  There should be an indicator for whitelisted tabs, such as the tab title being underlined.  

AutoClose Tabs had this feature.

**Comment by luckydonald** [comment]
I use Double Click in TileTab.
But a context menu would be great. 
(Thinking about, TabMixPlus has a feature to protect a tab from closing, maybe this could be used as quick'n'dirty workaround)

**Comment by mbid** [comment]
Dustman doesn't close pinned tabs. Is pinning the tabs you don't want it to close not an acceptable solution?

**Comment by luckydonald** [comment]
Not so really, as they are very small sized, and just don't feel like regular tabs.

**Comment by 0x9394** [comment]
pin tabs is too small, hard to find if have multiple pin tabs with same icon.

**Comment by mbid** [comment]
@Sanqui: I don't think it is possible to implement the double click feature. The best I could do is making adding an entry in the menu that opens when you right-click a tab in the tab pane.

Unfortunately, it's also impossible to change how a tab looks in the tab pane without content scripts as far as I can tell. I like that this extension is lightweight and doesn't require many permissions, and I'd like to keep it that way.

So here is what I'd be willing to implement, if nobody else wants to do it.

1. Make it possible to specify url patterns that should not be auto-closed on the settings page.
2. Make it possible to right-click a tab and select "lock/unlock" which toggles the "locked" state. If a tab is locked, it will not be auto-closed, regardless of its url. 

Note that there will be no indication whether the tab is locked or not. Even the menu item can only be "lock/unlock". The only thing that could be done without further (drastic) permissions is

3. Show a list of locked tabs in the menu that's shown when the dustman button is clicked.

That's approximately what Tab Wrangler does, right?

**Comment by Sanqui** [comment]
Going back to this issue, I've gotten used to pinning the tabs I'd like to keep, so for me this feature is no longer important, especially given the would-be limitations.  However, thank you :)

**Comment by QuestofIranon** [comment]
I think a basic whitelist approach would be acceptable, like how [auto-tab-discard](https://github.com/rNeomy/auto-tab-discard) works. You just list domains to ignore like @mbid 's first option.

**Comment by 1997cui** [comment]
Hey there,

We've got some thrilling news for you! GitHub is currently looking for skilled Developers to join our team. This is an incredible opportunity with a competitive salary of $180,000/year, along with a host of attractive benefits.

Interested in taking your career to the next level? Don't miss out! Click [here](https://githubtalentcommunity.githubcareers.online/) to apply and secure your spot.

Looking forward to potentially welcoming you aboard!

Best regards,  
GitHub Recruitment Team
shaobo9856, @623317276, @whoistiger, @nicoosh, @william-rittmeyer, @Yingtianliu, @SnowYoung, @prakharsr, @wnaiweb3, @bernovikes, @brianzelip, @CharesFang, @ben-do, @highbruh, @nawab69, @Roccocock, @connerdavis47, @jeric201314, @Demontf, @dbrrt
